js/pro/poloniex.js

```diff
@@ -0,0 +1,886 @@
+        /**
+         * @method
+         * @name poloniex#watchTrades
+         * @description get the list of most recent trades for a particular symbol
+         * @see https://docs.poloniex.com/#public-channels-market-data-trades
+         * @param {string} symbol unified symbol of the market to fetch trades for
+         * @param {int|undefined} since timestamp in ms of the earliest trade to fetch
+         * @param {int|undefined} limit the maximum amount of trades to fetch
+         * @param {object} params extra parameters specific to the poloniex api endpoint
+         * @returns {[object]} a list of [trade structures]{@link https://docs.ccxt.com/en/latest/manual.html?#public-trades}
+         */
+        await this.loadMarkets ();
+        const name = 'trades';
+        const trades = await this.subscribe (name, 'public', [ symbol ], params);
+        if (this.newUpdates) {
+            limit = trades.getLimit (symbol, limit);
+        }
+        return this.filterBySinceLimit (trades, since, limit, 'timestamp', true);
+    }
+
+    async watchOrderBook (symbol, limit = undefined, params = {}) {
+        /**
+         * @method
+         * @name poloniex#watchOrderBook
+         * @description watches information on open orders with bid (buy) and ask (sell) prices, volumes and other data
+         * @see https://docs.poloniex.com/#public-channels-market-data-book-level-2
+         * @param {string} symbol unified symbol of the market to fetch the order book for
+         * @param {int|undefined} limit not used by poloniex watchOrderBook
+         * @param {object} params extra parameters specific to the poloniex api endpoint
+         * @returns {object} A dictionary of [order book structures]{@link https://docs.ccxt.com/en/latest/manual.html#order-book-structure} indexed by market symbols
+         */
+        const name = 'book_lv2';
+        const orderbook = await this.subscribe (name, 'public', [ symbol ], params);
+        return orderbook.limit ();
+    }
+
+    async watchOrders (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        /**
+         * @method
+         * @name poloniex#watchOrders
+         * @description watches information on multiple orders made by the user
+         * @see https://docs.poloniex.com/#authenticated-channels-market-data-orders
+         * @param {string|undefined} symbol unified market symbol of the market orders were made in
+         * @param {int|undefined} since not used by poloniex watchOrders
+         * @param {int|undefined} limit not used by poloniex watchOrders
+         * @param {object} params extra parameters specific to the poloniex api endpoint
+         * @returns {[object]} a list of [order structures]{@link https://docs.ccxt.com/en/latest/manual.html#order-structure}
+         */
+        await this.loadMarkets ();
+        const name = 'orders';
+        const authentication = this.authenticate ();
```

Comment:
as I said authenticate is not correctly implemented and besides that you're calling an async method without awaiting it.